pkg/controller/database/dbsubnetgroup/controller.go

```diff
@@ -18,175 +18,206 @@ package dbsubnetgroup
+	}
 
 	return managed.ExternalObservation{
-		ResourceExists:    true,
-		ConnectionDetails: managed.ConnectionDetails{},
+		ResourceUpToDate: upToDate,
+		ResourceExists:   true,
 	}, nil
 }
 
 func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
-	cr, ok := mgd.(*v1alpha3.DBSubnetGroup)
+	cr, ok := mgd.(*v1beta1.DBSubnetGroup)
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
 	}
 
-	cr.Status.SetConditions(runtimev1alpha1.Creating())
+	cr.SetConditions(runtimev1alpha1.Creating())
 
-	input := &awsrds.CreateDBSubnetGroupInput{
-		DBSubnetGroupDescription: aws.String(cr.Spec.DBSubnetGroupDescription),
-		DBSubnetGroupName:        aws.String(cr.Spec.DBSubnetGroupName),
-		SubnetIds:                cr.Spec.SubnetIDs,
-		Tags:                     []awsrds.Tag{},
+	if cr.Status.AtProvider.SubnetGroupStatus == v1beta1.DBSubnetGroupStateCreating {
+		return managed.ExternalCreation{}, nil
 	}
 
-	for _, t := range cr.Spec.Tags {
-		input.Tags = append(input.Tags, awsrds.Tag{
-			Key:   aws.String(t.Key),
-			Value: aws.String(t.Value),
-		})
+	input := &awsrds.CreateDBSubnetGroupInput{
+		DBSubnetGroupDescription: aws.String(cr.Spec.ForProvider.DBSubnetGroupDescription),
+		DBSubnetGroupName:        aws.String(meta.GetExternalName(cr)),
+		SubnetIds:                cr.Spec.ForProvider.SubnetIDs,
+		Tags:                     []awsrds.Tag{},
 	}
 
-	req := e.client.CreateDBSubnetGroupRequest(input)
-
-	response, err := req.Send(ctx)
+	_, err := e.client.CreateDBSubnetGroupRequest(input).Send(ctx)
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+}
 
-	if err != nil {
-		return managed.ExternalCreation{}, errors.Wrapf(err, errCreate, cr.Spec.DBSubnetGroupName)
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1beta1.DBSubnetGroup)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
 	}
 
-	cr.UpdateExternalStatus(*response.DBSubnetGroup)
-
-	return managed.ExternalCreation{}, nil
-}
+	if cr.Status.AtProvider.SubnetGroupStatus == v1beta1.DBSubnetGroupStateModifying {
+		return managed.ExternalUpdate{}, nil
+	}
 
-func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
-	// TODO(soorena776): add more sophisticated Update logic, once we
-	// categorize immutable vs mutable fields (see #727)
+	_, err := e.client.ModifyDBSubnetGroupRequest(&awsrds.ModifyDBSubnetGroupInput{
+		DBSubnetGroupName:        aws.String(meta.GetExternalName(cr)),
+		DBSubnetGroupDescription: aws.String(cr.Spec.ForProvider.DBSubnetGroupDescription),
+		SubnetIds:                cr.Spec.ForProvider.SubnetIDs,
+	}).Send(ctx)
 
-	return managed.ExternalUpdate{}, nil
+	return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(dbsg.IsErrorAlreadyExists, err), errUpdate)
```

Comment:
It's possible to update the tags as well but it requires another type of call `AddTagsToResourceRequest`, you can take a look at RDS controller for an example. In that case, a patch is used but here you can just go with `cr.Spec.ForProvider.Tags` to calculate the list of tags to send. Comments in `AddTagsToResourceRequest` call says it applies to RDS resources but it looks like [DBSubnetGroup is also considered](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Tagging.html) an RDS resource.